File: ios/Runner/Extension/AlbumModelExtension.swift
//
//  AlbumModelExtension.swift
//  Runner
//
//  Created by t k on 2024/08/14.
//

extension AlbumModelContract {
    func toPigeon() -> AlbumDTO {
        return AlbumDTO(
            id: String(self.id.raw),
            name: self.name,
            artistId: String(self.artistId.raw),
            artistName: self.artistName,
            imageId: self.imageId
        )
    }
}

File: android/app/src/main/kotlin/dev/tcode/thinmpf/extension/AlbumModelExtension.kt
package dev.tcode.thinmpf.extension

import dev.tcode.thinmpf.model.contract.AlbumModelContract
import dev.tcode.thinmpf.pigeon_output.AlbumDTO

fun AlbumModelContract.toPigeon(): AlbumDTO {
    return AlbumDTO(
        id = this.id.raw,
        name = this.name,
        artistId = this.artistId.raw,
        artistName = this.artistName,
        imageId = this.imageId
    )
}
